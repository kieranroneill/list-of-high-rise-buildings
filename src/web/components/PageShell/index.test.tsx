import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';

// Components.
import { PageShell, Props } from './';

interface Scope {
    props: Props;
    wrapper: ShallowWrapper;
}

describe('src/web/components/PageShell', () => {
    let scope: Scope;

    beforeEach(() => {
        const props: Props = {
            children: <h1>Hello human</h1>,
            getCities: jest.fn(),
        };

        scope = {
            props,
            wrapper: shallow(<PageShell {...props} />),
        };
    });

    describe('<PageShell /> snapshots', () => {
        it('should match the snapshot with the default props', () => {
            expect(scope.wrapper).toMatchSnapshot();
        });
    });

    describe('when the component mounts', () => {
        it('should dispatch an action to get the cities', () => {
            expect(scope.props.getCities).toBeCalled();
        });
    });
});
