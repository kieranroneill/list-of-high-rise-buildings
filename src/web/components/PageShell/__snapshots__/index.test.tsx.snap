// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/web/components/PageShell <PageShell /> snapshots should match the snapshot with the default props 1`] = `
<styled.div>
  <styled.div>
    <WithStyles(Typography)
      gutterBottom={true}
      variant="h3"
    >
      List of high rise buildings
    </WithStyles(Typography)>
    <WithStyles(Paper)>
      <styled.div>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          #
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          City
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          Country
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          All Buildings
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          100m+
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          150m+
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          200m+
        </WithStyles(Button)>
        <WithStyles(Button)
          color="primary"
          component={[Function]}
          style={
            Object {
              "margin": "0.5rem",
            }
          }
          variant="contained"
        >
          300m+
        </WithStyles(Button)>
      </styled.div>
    </WithStyles(Paper)>
  </styled.div>
  <h1>
    Hello human
  </h1>
</styled.div>
`;
